### PR TITLE
additional minor updates to CMS_*_ContentType documentation

### PR DESCRIPTION
**Pull request by mcr**
##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

Building upon previous two submissions, and after understanding more of what things do, added some more clarifying text.  Some of which might be wrong.

### CODE REVIEW

File: doc/man3/CMS_get0_type.pod [hunk 1]
```diff
@@ -26,10 +27,14 @@ flag and B<before> the structure is finalised, otherwise the results are
...
-content type.
+content type.  It differs from CMS_get0_type in that it points to the actual
+eContentType element contained in the ASN1 structure that was constructed or received.
+It should return a single encoded OID.
 
 CMS_get0_content() returns a pointer to the B<ASN1_OCTET_STRING> pointer
-containing the embedded content.
+containing the embedded content itself.
+It's use is undefined on outgoing objects created with B<CMS_PARTIAL>
+until L<CMS_final> is called.
```
Comment (mcr) [inline doc/man3/CMS_get0_type.pod]: This statement might be incorrect.  I'm unclear what the API contract should be here.

File: doc/man3/CMS_get0_type.pod [hunk 2]
```diff
@@ -15,20 +15,26 @@ CMS_get0_type, CMS_set1_eContentType, CMS_get0_eContentType, CMS_get0_content -
...
-and ASN1_OBJECT pointer. An application can then decide how to process the
+The CMS_get0_type() function returns a pointer to the content type of a CMS_ContentInfo structure as
+an ASN1_OBJECT pointer.  This is typically set when the CMS_ContentInfo structure is created, or when a signed or encrypted artifact is decoded.
+An application can then decide how to process the
 CMS_ContentInfo structure based on this value.
 
 CMS_set1_eContentType() sets the embedded content type of a CMS_ContentInfo
-structure. It should be called with CMS functions with the B<CMS_PARTIAL>
+structure. It should be called with CMS functions (such as L<CMS_sign>, L<CMS_encrypt>)
+with the B<CMS_PARTIAL>
```
Comment (mattcaswell) [inline doc/man3/CMS_get0_type.pod]: This change seems to be the same one as in #7960. That commit should probably be dropped from this PR.

File: doc/man3/CMS_get0_type.pod [hunk 3]
```diff
@@ -15,20 +15,26 @@ CMS_get0_type, CMS_set1_eContentType, CMS_get0_eContentType, CMS_get0_content -
...
 flag and B<before> the structure is finalised, otherwise the results are
 undefined.
 
 ASN1_OBJECT *CMS_get0_eContentType() returns a pointer to the embedded
-content type.
+content type.  It differs from CMS_get0_type in that it points to the actual
+eContentType element contained in the ASN1 structure that was constructed or received.
```
Comment (mattcaswell) [inline doc/man3/CMS_get0_type.pod]: Somehow this description wasn't quite enough for me to understand what it did. I still had to check the source. Perhaps an example would help?

File: doc/man3/CMS_get0_type.pod (hunk 1 above)
Comment (mattcaswell) [reply doc/man3/CMS_get0_type.pod]: I'm not sure that statement is correct. AFAICT (from code inspection only), I think you will get back a valid \*\* pointer, but then thing it points to will be NULL, indicating that the embedded content has not been created yet.

**Comment by mcr** [comment]
I don't seem to be able to reply inline to Matt's comments.
I have (a) rebased this which has automatically removed the content duplicated in #7960.
(b) I no longer have much memory of the difference between get0_type() and get0_eContentType(), so I would have to go back to the source code.  I am willing to generate an example, but I don't really know how much context to put around the example.
Can you point me at another documentation example which is at the same level of complexity?

**Comment by mattcaswell** [comment]
>  I no longer have much memory of the difference between get0_type() and get0_eContentType(), so I would have to go back to the source code. I am willing to generate an example, but I don't really know how much context to put around the example.

I think all I really meant here was to provide some examples of content types that you might set here, e.g. IIUC it could be the OID returned by  `OBJ_obj2txt("id-ct-asciiTextWithCRLF", 0)` or `OBJ_obj2txt("id-ct-xml", 0)`

**Pull request closed without merging**